src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientMethodsAnalyzer.cs

```diff
@@ -279,6 +279,60 @@ private static bool IsClientMethodReturnType(ISymbolAnalysisContext context, IMe
             return false;
         }
 
+        private static bool DoReturnTypesMatch(ITypeSymbol asyncReturnType, ITypeSymbol syncReturnType)
+        {
+            if (asyncReturnType == null || syncReturnType == null)
+            {
+                return false;
+            }
+
+            // If the async return type is non-generic Task
+            if (asyncReturnType.Name == TaskTypeName && !((INamedTypeSymbol)asyncReturnType).IsGenericType)
```

Comment:
```suggestion
            if (asyncReturnType.Name == TaskTypeName && (!((INamedTypeSymbol)asyncReturnType).IsGenericType))
```

I'd recommend explicitly scoping anything with the not operator, as it can get weird in compound clauses otherwise.